File: backend/app/schemas/review.py
# schemas/review.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime

class ReviewCreate(BaseModel):
    movie_id: int
    rating: Optional[int] = Field(None, ge=1, le=10)
    comment: Optional[str] = None
    is_spoiler: bool = False

class ReviewUpdate(BaseModel):
    rating: Optional[int] = Field(None, ge=1, le=10)
    comment: Optional[str] = None
    is_spoiler: Optional[bool] = None

class ReviewOut(BaseModel):
    id: int
    user_id: int  
    movie_id: int
    rating: Optional[int] = None
    comment: Optional[str]
    is_spoiler: bool
    created_at: datetime
    
    class Config:
        from_attributes = True

class ReviewModerateUpdate(BaseModel):
    comment: Optional[str] = None
    is_spoiler: Optional[bool] = None

File: android/app/src/main/java/com/app/movieit/data/model/Review.kt
package com.app.movieit.data.model

import com.google.gson.annotations.SerializedName

data class ReviewCreate(
    @SerializedName("movie_id") val movieId: Int,
    val rating: Int? = null,
    val comment: String? = null,
    @SerializedName("is_spoiler") val isSpoiler: Boolean = false
)

data class ReviewUpdate(
    val rating: Int? = null,
    val comment: String? = null,
    @SerializedName("is_spoiler") val isSpoiler: Boolean? = null
)

data class ReviewOut(
    val id: Int,
    @SerializedName("user_id") val userId: Int,
    @SerializedName("movie_id") val movieId: Int,
    val rating: Int?,
    val comment: String?,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("is_spoiler") val isSpoiler: Boolean
)

data class ReviewModerateUpdate(
    val comment: String? = null,
    @SerializedName("is_spoiler") val isSpoiler: Boolean? = null
)